hazelcast/include/hazelcast/client/decimal.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <vector>
+#include <boost/multiprecision/cpp_int.hpp>
+#include "hazelcast/util/export.h"
+
+#if defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
+#pragma warning(push)
+#pragma warning(disable : 4251) // for dll export
+#endif
+
+namespace hazelcast {
+namespace client {
+/**
+ *  An arbitrary precision and scale floating point number.
+ *  unscaledValue x 10 ^ -scale
+ *
+ *  For arithmetic operations support, it is suggested to use external
+ *  libraries. An usage example with boost::multiprecision::cpp_dec_float
+ *  could be as follows:
+ *  <pre><code>
+ * hazelcast::decimal dec{ u, 2 };
+ * boost::multiprecision::cpp_dec_float<10> f(
+ *  (dec.unscaled.str() + "e-" + std::to_string(dec.scale)).c_str());
+ *  std::cout << f.str(100, std::ios_base::dec) << std::endl;
+ *  </code></pre>
+ */
+struct HAZELCAST_API decimal
+{
+    boost::multiprecision::cpp_int unscaled;
+    int32_t scale;
+};
+
+namespace pimpl {
+
+/**
+ * Takes twos complement of giving array where most significant value is first.
+ * @param a the array to take twos complement of
+ */
+
+void HAZELCAST_API
```

Comment:
What does HAZELCAST_API mean? As I know it makes an API public. I think this method should be internal.

Reply:
HAZELCAST_API does not mean public. It is visibility modifiers so that Windows will compile. Anything defined in the headers should have this as far as I remember.